### PR TITLE
[4.2] Change tags plugin to service provider

### PR DESCRIPTION
**Pull request by laoneo**
### Summary of Changes
Change tags plugin to service provider. Did also some code style cleanup.

### Testing Instructions
- Open the article form
- Define some tags
- Save the article

### Actual result BEFORE applying this Pull Request
Tags are stored and assigned to the new article.

### Expected result AFTER applying this Pull Request
Tags are stored and assigned to the new article.

### CODE REVIEW

File: plugins/behaviour/taggable/src/Extension/Taggable.php [hunk 1]
```diff
@@ -23,39 +33,43 @@
  *
  * @since  4.0.0
  */
-class PlgBehaviourTaggable extends CMSPlugin
+class Taggable extends CMSPlugin implements SubscriberInterface
```
Comment (heelc29) [inline plugins/behaviour/taggable/src/Extension/Taggable.php]: Should be the class `final`? https://github.com/joomla/joomla-cms/pull/36042#discussion_r748905822

File: plugins/behaviour/taggable/taggable.xml [hunk 2]
```diff
@@ -9,8 +9,10 @@
 	<copyright>(C) 2016 Open Source Matters, Inc.</copyright>
 	<license>GNU General Public License version 2 or later; see LICENSE.txt</license>
 	<description>PLG_BEHAVIOUR_TAGGABLE_XML_DESCRIPTION</description>
+	<namespace path="src">Joomla\Plugin\Behaviour\Taggable</namespace>
 	<files>
-		<filename plugin="taggable">taggable.php</filename>
+		<folder plugin="taggable">services</folder>
+		<folder>src</folder>
 	</files>
```
Comment (heelc29) [inline plugins/behaviour/taggable/taggable.xml:16]: Can you add the `languages` to the manifest?

File: plugins/behaviour/taggable/src/Extension/Taggable.php [hunk 3]
```diff
@@ -325,13 +339,13 @@ public function onTableAfterLoad(CmsEvent\Table\AfterLoadEvent $event)
 	/**
 	 * Runs when an existing table object has been loaded
 	 *
-	 * @param   CmsEvent\Model\BeforeBatchEvent $event The event to handle
+	 * @param   BeforeBatchEvent $event The event to handle
```
Comment (heelc29) [inline plugins/behaviour/taggable/src/Extension/Taggable.php]:
```suggestion
	 * @param   BeforeBatchEvent  $event  The event to handle
```

File: plugins/api-authentication/token/src/Extension/Token.php [hunk 4]
```diff
@@ -34,7 +34,7 @@ final class Token extends CMSPlugin
 	/**
 	 * The application object
 	 *
-	 * @var    ApiApplication
+	 * @var    CMSApplicationInterface
```
Comment (heelc29) [inline plugins/api-authentication/token/src/Extension/Token.php:37]: <s>Correct Import</s>
@laoneo Can you please adapt this to the `CMSApplicationInterface` as well? I think I didn't explain clearly what I mean 😅.
https://github.com/joomla/joomla-cms/blob/c0dcbf2dbf9924a661061b2494ff917e9dbc5d8c/plugins/api-authentication/token/src/Extension/Token.php#L14

File: plugins/behaviour/taggable/src/Extension/Taggable.php [hunk 5]
```diff
@@ -325,13 +339,13 @@ public function onTableAfterLoad(CmsEvent\Table\AfterLoadEvent $event)
...
+	 * @param   BeforeBatchEvent  $event  The event to handle
 	 *
 	 * @return  void
 	 *
 	 * @since   4.0.0
 	 */
-	public function onBeforeBatch(CmsEvent\Model\BeforeBatchEvent $event)
+	public function onBeforeBatch(BeforeBatchEvent $event)
 	{
 		/** @var TableInterface $oldTable */
```
Comment (heelc29) [inline plugins/behaviour/taggable/src/Extension/Taggable.php]:
```suggestion
		/** @var TableInterface $sourceTable */
```

**Comment by joomdonation** [comment]
I have tested this item :white_check_mark: successfully on 00fc4461757d818c34e46e30d447e653ede19e90<br /><br />Add, remove tags, batch assign tags, all still works as expected<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37928">issues.joomla.org/tracker/joomla-cms/37928</a>.</sub>

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 00fc4461757d818c34e46e30d447e653ede19e90<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37928">issues.joomla.org/tracker/joomla-cms/37928</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37928">issues.joomla.org/tracker/joomla-cms/37928</a>.</sub>

**Comment by roland-d** [comment]
Thanks everybody

**Pull request merged**